### ISSUE TITLE
Feature Request: Widget Background Colors

### ISSUE THREAD

**Issue by cgarwood** [resoai/TileBoard#13]
It would be great to be able to define background colors for widgets with a function or state map. For example, if a door sensor is `on` or `open` have a red background, as opposed to a green background if it is `off` or `closed`

Likewise, being able to simply set background colors for different entities would be useful, for example, color coding light widgets based on room or type

**Comment by resoai** [comment]
You can kind of do that with `classes: ["-your-class"]`

**Comment by cgarwood** [comment]
Looks like that would work for the second half of the request, just setting the static background color. Is there a way to programatically add or remove classes based on the state? Keeping with the classes is probably better than manually specifying a background-color in the widget config

**Comment by Evgeny-** [comment]
@cgarwood not sure if it's enough, but we already adding class to the element by it's state.  E.g. is state is "off" class will be "-off".

Is it something you looked in?

**Comment by cgarwood** [comment]
That's awesome and covers 90% of my use-cases. The only use case that I don't think would be covered well by that is a numeric sensor like a temperature sensor, where it might be nice to change the background color on a sliding scale from blue to green to red depending on how cold or hot it is, with the shade of the color being determined by the temperature. That would be a lot of CSS classes to define, since you'd have to define one for each possible temperature, which would add up quickly for sensors that report decimals.

**Comment by cgarwood** [comment]
One other use case I can think of that's not covered:

My Nest thermostat creates a `climate` entity, whose state is the current operation mode of the thermostat (off, eco, heat, cool, auto). I'd like to change the background color of the widget depending on if the unit is actively heating (orange), cooling (blue), or idle (normal). Unfortunately, that information isn't reflected in the state of the climate entity (and in fact, isn't even included in the attributes, it's a separate `sensor` entity in HomeAssistant). It provides a separate `sensor` entity that returns off, heating, or cooling depending on what the unit is actively doing.

I could see that being more difficult though, as it requires being able to grab the state of a different entity

**Comment by resoai** [comment]
You can access the state of a different entity like so:

```
state: function () {
    return this.states['sensor.other_entity'].state;
 }
```

You could create separate css class for each of those states. I'm sure we can work something with the background in future but we do need a better approach than simply changing the background.

**Comment by resoai** [comment]
@cgarwood You can now do what you wanted :)
  
```
{
                     position: [2, 3],
                     type: TYPES.FAN,
                     title: 'Ceiling fan',
                     id: 'fan.ceiling_fan',
                     customCss: {
                        'background-color': 'pink'
                     }
}
```

**Comment by resoai** [comment]
@cgarwood could you please confirm it works the way you wanted to?

**Comment by cgarwood** [comment]
Looks like `styles:` isn't quite working right. It's overriding the positional (and other) styles on tiles.
```
{
	position: [0, 0],
	width: 1,
	type: TYPES.LIGHT,
	id: "light.living_room",
	title: 'Living Room',
	subtitle: 'Lights',
	icon: 'mdi-lightbulb'
},
```
gives me:
```
element.style {
    width: 140px;
    height: 140px;
    left: 0px;
    top: 0px;
}
```

while
```
{
	position: [0, 0],
	width: 1,
	type: TYPES.LIGHT,
	id: "light.living_room",
	title: 'Living Room',
	subtitle: 'Lights',
	icon: 'mdi-lightbulb',
	styles: {
		'background-color' : 'pink'
	}
},
```
only gives me:
```
element.style {
    background-color: pink;
}
```

all of the positional styles are overridden and removed

**Comment by cgarwood** [comment]
The only other thing I'd like to be able to do with it is use a function to return the value of a style, so I could write a function to take the numeric value of a temperature sensor and adjust the background color along a gradient of orange to blue depending on the sensor value.

**Comment by resoai** [comment]
Can you please try now like this:

{
	position: [0, 0],
	width: 1,
	type: TYPES.LIGHT,
	id: "light.living_room",
	title: 'Living Room',
	subtitle: 'Lights',
	icon: 'mdi-lightbulb',
	customStyles: {
		'background-color' : 'pink'
	}
}

**Comment by cgarwood** [comment]
tested today and it's working now

**Comment by resoai** [comment]
Great! Does this cover your use case?

**Comment by resoai** [comment]
Do you think we should setup a telegram channel to discuss things? This could be a bit more practical in certain occasions.

**Comment by cgarwood** [comment]
Mostly covers my use cases. I've managed to "abuse" anonymous functions in the `state:` line of a tile config to cover the temperature-to-background-color conversion. Though if the state function returns false, '', etc it shows the function() { code on the UI, so I just have it returning a space ' ' right now. But yes, this enables me to do what I wanted :)

Some kind of chat could be beneficial, I've noticed a lot of ideas/feature requests come form the same handful of people so there's definitely a group of power users that could benefit from it.
